backend-api/src/functions/asyncActiveSession/tokenService/tokenService.ts

```diff
@@ -52,6 +51,38 @@ export class TokenService implements ITokenService {
     return successResult("");
   };
 
+  private async fetchPublicKeyWithRetries(
+    stsJwksEndpoint: string,
+  ): Promise<Result<JSON>> {
+    const maxRetries = 2;
+    const delayInMs = 1000;
+
+    let fetchPublicKeyResult: Result<JSON> | undefined;
+
+    for (let retries = 0; retries <= maxRetries; retries++) {
+      fetchPublicKeyResult = await this.fetchPublicKey(stsJwksEndpoint);
+
+      if (!fetchPublicKeyResult.isError) {
+        return fetchPublicKeyResult;
+      }
+
+      if (retries < maxRetries) {
+        await new Promise((resolve) => setTimeout(resolve, delayInMs));
+      }
+    }
+
+    // After all retries, check if fetchPublicKeyResult was assigned and handle the case where it was not assigned (should not happen)
+    if (fetchPublicKeyResult == null) {
+      return errorResult({
```

Comment:
Not tested but not sure it's worth as it shouldn't ever happen, this is to satisify a satisfy. Does anyone:
- Know a way to avoid this check?
- If not, think we should still unit test it? Would we have to mock the `fetchPublicKey` func (which is currently readonly)

Reply:
Thinking about the above ^ I'd almost recommend casting the .json() response to `unknown`. This means the compiler will explicitly force us to check the structure of the response. `any` is a way of telling the compiler: 'Trust me - I know what I'm doing and I can perform any method on this mystery variable'

Reply:
Feels counter intuitive but would it be better to do response.text() and then parse it so we know it's JSON? Or do we make a public key type that we validate it's the shape of?